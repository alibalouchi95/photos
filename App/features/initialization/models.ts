// The initialization state of the Textile instance
// Uninitialized: before an account is synced/created
// Initializing: the user has initiated the account sync/creation process
// Initialized: the instance is initialized as an account peer
export enum TextileInstanceState {
  uninitialized = 'uninitialized',
  initializing = 'initializing',
  initialized = 'initialized'
}
