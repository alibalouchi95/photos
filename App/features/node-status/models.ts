export enum NodeState {
  stopped = 'stopped',
  started = 'started',
  online = 'online'
}
