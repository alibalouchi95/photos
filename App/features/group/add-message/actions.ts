import { createAsyncAction } from 'typesafe-actions'

export interface addMessagePayload {
  id: string
  groupId: string
}

export interface addMessageRequestPayload extends addMessagePayload {
  body: string
}

export interface addMessageFailurePayload extends addMessagePayload {
  error: any
}

export const addMessage = createAsyncAction(
  'group/add-message/ADD_MESSAGE_REQUEST',
  'group/add-message/ADD_MESSAGE_SUCCESS',
  'group/add-message/ADD_MESSAGE_FAILURE'
)<addMessageRequestPayload, addMessagePayload, addMessageFailurePayload>()
