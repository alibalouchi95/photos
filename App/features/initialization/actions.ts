import { createAction } from 'typesafe-actions'
import { OnboardingPath } from './models'

export const initializeNewAccount = createAction(
  'initialization/INITIALIZE_CREATING_NEW_WALLET_AND_ACCOUNT'
)

export const initializeExistingAccount = createAction(
  'initialization/INITIALIZE_WITH_EXISTING_ACCOUNT_SEED',
  resolve => {
    return (seed: string) => resolve({ seed })
  }
)

export const nodeInitialized = createAction('initialization/NODE_INITIALIZED')

export const failedToInitializeNode = createAction(
  'initialization/FAILED_TO_INITIALIZE_NODE',
  resolve => {
    return (error: any) => resolve({ error })
  }
)

export const chooseOnboardingPath = createAction(
  'initialization/CHOOSE_ONBOARDING_PATH',
  resolve => {
    return (path: OnboardingPath) => resolve({ path })
  }
)

export const setCurrentPage = createAction(
  'initialization/SET_CURRENT_PAGE_ONBOARDING',
  resolve => {
    return (page: number) => resolve({ page })
  }
)

export const nextPage = createAction('initialization/NEXT_PAGE_ONBOARDING')

export const onboardingSuccess = createAction(
  'initialization/COMPLETE_ONBOARDING'
)
