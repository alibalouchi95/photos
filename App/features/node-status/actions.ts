import { createAction } from 'typesafe-actions'

export const failedToInitializeNode = createAction(
  'node-status/FAILED_TO_INITIALIZE_NODE',
  resolve => (error: any) => resolve({ error })
)

export const nodeStarted = createAction('node-status/NODE_STARTED')

export const nodeStopped = createAction('node-status/NODE_STOPPED')

export const nodeOnline = createAction('node-status/NODE_ONLINE')

export const nodeFailedToStart = createAction(
  'node-status/NODE_FAILED_TO_START',
  resolve => (error: string) => resolve({ error })
)

export const nodeFailedToStop = createAction(
  'node-status/NODE_FAILED_TO_STOP',
  resolve => (error: string) => resolve({ error })
)

export const stopNodeAfterDelayStarting = createAction(
  'node-status/STOP_NODE_AFTER_DELAY_STARTING',
  resolve => (delay: number) => resolve({ delay })
)

export const stopNodeAfterDelayCancelled = createAction(
  'node-status/STOP_NODE_AFTER_DELAY_CANCELLED'
)
