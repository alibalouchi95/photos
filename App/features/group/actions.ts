import { feedActions as feed } from './feed'
import { addMessageActions as addMessage } from './add-message'
import { addPhotoActions as addPhoto } from './add-photo'
import { renameGroupActions as renameGroup } from './rename-group'
import { ignoreActions as ignore } from './ignore'

export { feed, addMessage, addPhoto, renameGroup, ignore }
