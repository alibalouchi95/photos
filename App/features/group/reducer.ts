import { combineReducers } from 'redux'
import { ActionType } from 'typesafe-actions'

import { feedReducer, FeedState } from './feed'
import { addMessageReducer, AddMessageState } from './add-message'
import { addPhotoReducer, ProcessingImagesState } from './add-photo'
import { renameGroupReducer, RenameGroupState } from './rename-group'

import * as actions from './actions'

export interface GroupState {
  feed: FeedState
  addMessage: AddMessageState
  addPhoto: ProcessingImagesState
  renameGroup: RenameGroupState
}

export type GroupAction = ActionType<typeof actions>

export default combineReducers<GroupState>({
  feed: feedReducer,
  addMessage: addMessageReducer,
  addPhoto: addPhotoReducer,
  renameGroup: renameGroupReducer
})
