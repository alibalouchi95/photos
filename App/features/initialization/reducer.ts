import { combineReducers } from 'redux'
import { ActionType, getType } from 'typesafe-actions'

import { TextileInstanceState } from './models'
import * as actions from './actions'

// When the path changes, currentPage should be reset to 0
export interface InitializationState {
  readonly onboarding: {
    completed: boolean
    path: string
    currentPage: number
  }
  readonly instance: {
    readonly state: TextileInstanceState
    readonly error?: string
  }
}

export type InitializationAction = ActionType<typeof actions>

export default combineReducers<InitializationState, InitializationAction>({
  onboarding: (
    state = {
      completed: false,
      path: 'default',
      currentPage: 0
    },
    action
  ) => {
    switch (action.type) {
      case getType(actions.chooseOnboardingPath): {
        return {
          ...state,
          path: action.payload.path,
          currentPage: 0
        }
      }
      case getType(actions.nextPage): {
        return {
          ...state,
          currentPage: state.currentPage + 1
        }
      }
      case getType(actions.setCurrentPage): {
        return {
          ...state,
          currentPage: action.payload.page
        }
      }
      case getType(actions.onboardingSuccess): {
        return {
          ...state,
          completed: true
        }
      }
      default:
        return state
    }
  },
  instance: (
    state = {
      state: TextileInstanceState.uninitialized
    },
    action
  ) => {
    switch (action.type) {
      case getType(actions.initializeNewAccount):
      case getType(actions.initializeExistingAccount): {
        return {
          state: TextileInstanceState.initializing
        }
      }
      case getType(actions.failedToInitializeNode): {
        const { error } = action.payload
        const message =
          (error.message as string) || (error as string) || 'unknown error'
        return {
          state: TextileInstanceState.uninitialized,
          error: message
        }
      }
      case getType(actions.nodeInitialized): {
        return {
          state: TextileInstanceState.initialized
        }
      }
      default:
        return state
    }
  }
})
