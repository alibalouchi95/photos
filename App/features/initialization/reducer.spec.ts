import * as actions from './actions'
import reducer from './reducer'
import { InitializationStatus } from './models'

const initialState = reducer(undefined, {} as any)

const initializationError = 'error'

describe('initialization', () => {
  describe('initial state', () => {
    it('should match snapshot', () => {
      expect(initialState).toMatchSnapshot()
    })
  })
  describe('textile account initialization', () => {
    it('should initialize a new account', () => {
      const state0 = reducer(initialState, actions.initializeNewAccount())
      const creatingWallet: InitializationStatus = 'creatingWallet'
      expect(state0.instance.state).toEqual(creatingWallet)
      const state1 = reducer(
        state0,
        actions.updateInitializationStatus('initialized')
      )
      const initialized: InitializationStatus = 'initialized'
      expect(state1.instance.state).toEqual(initialized)
    })
    it('should log an error initializing a new account', () => {
      const state0 = reducer(initialState, actions.initializeNewAccount())
      const creatingWallet: InitializationStatus = 'creatingWallet'
      expect(state0.instance.state).toEqual(creatingWallet)
      const state1 = reducer(
        state0,
        actions.failedToInitializeNode(initializationError)
      )
      expect(state1.instance.error).toEqual(initializationError)
    })
  })
})
