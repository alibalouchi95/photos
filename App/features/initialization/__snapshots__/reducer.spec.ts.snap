// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialization initial state should match snapshot 1`] = `
Object {
  "instance": Object {
    "state": "uninitialized",
  },
  "onboarding": Object {
    "completed": false,
    "currentPage": 0,
    "path": 0,
  },
}
`;
